lms/djangoapps/course_api/v0/views.py

```diff
@@ -0,0 +1,451 @@
+        content_uri = request.build_absolute_uri(reverse('course_api:v0:content:detail',
+                                                         kwargs={'course_id': unicode(course_key),
+                                                                 'content_id': content_id}))
+
+    data['id'] = unicode(content_id)
+    data['uri'] = content_uri
+
+    # Include any additional fields requested by the caller
+    include_fields = request.QUERY_PARAMS.get('include_fields', None)
+    if include_fields:
+        include_fields = include_fields.split(',')
+        for field in include_fields:
+            data[field] = getattr(content_descriptor, field, None)
+
+    return data
+
+
+def _serialize_content_children(request, course_key, children):
+    """
+    Loads the specified content child data into the response dict
+    This should probably evolve to use DRF serializers
+    """
+    data = []
+    if children:
+        for child in children:
+            child_data = _serialize_content(
+                request,
+                course_key,
+                child
+            )
+            data.append(child_data)
+    return data
+
+
+def _serialize_content_with_children(request, course_key, descriptor, depth):  # pylint: disable=invalid-name
+    """
+    Serializes course content and then dives into the content tree,
+    serializing each child module until specified depth limit is hit
+    """
+    data = _serialize_content(
+        request,
+        course_key,
+        descriptor
+    )
+
+    if depth > 0:
+        data['children'] = []
+        for child in descriptor.get_children():
+            data['children'].append(_serialize_content_with_children(request, course_key, child, depth - 1))
+    return data
+
+
+def _get_course_data(request, course_key, course_descriptor, depth=0):
+    """
+    creates a dict of course attributes
+    """
+
+    if depth > 0:
+        data = _serialize_content_with_children(
+            request,
+            course_key,
+            course_descriptor,  # Primer for recursive function
+            depth
+        )
+        data['content'] = data['children']
+        data.pop('children')
+    else:
+        data = _serialize_content(
+            request,
+            course_key,
+            course_descriptor
+        )
+
+    data['image_url'] = ''
+    if getattr(course_descriptor, 'course_image'):
+        data['image_url'] = course_image_url(course_descriptor)
+
+    return data
+
+
+class CourseContentList(APIViewWithPermissions):
+    """
+    **Use Case**
+
+        CourseContentList gets a collection of content for a given
+        course. You can use the **uri** value in
+        the response to get details for that content entity.
+
+        The optional **depth** parameter that allows clients to get child content down to the specified tree level.
+
+    **Example requests**:
+
+        GET /api/courses/v0/{course_id}/content/
```

Comment:
I don't like including the full path in the docstring here, because the actual definition of the path is scattered across many modules and is not in any way inherently tied to this class.